demos/minimal/actors/sample_generator.py

```diff
@@ -1,68 +1,78 @@
 from improv.actor import Actor, RunManager
-from datetime import date  # used for saving
 import numpy as np
 import logging
+import time  # Importing time module for the delay
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 
 class Generator(Actor):
-    """Sample actor to generate data to pass into a sample processor.
-
-    Intended for use along with sample_processor.py.
+    """Sample actor to generate a sine/cosine wave based on frame number to pass into a sample processor. Odd frames generate a sine wave and even frames generate a cosine wave.
     """
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.data = None
         self.name = "Generator"
-        self.frame_num = 0
+        self.data = None
+        self.max_frames = 500  # Set the limit for number of frames
 
     def __str__(self):
         return f"Name: {self.name}, Data: {self.data}"
 
     def setup(self):
-        """Generates an array that serves as an initial source of data.
+        """Initializes all class variables.
 
-        Initial array is a 100 row, 5 column numpy matrix that contains
-        integers from 1-99, inclusive.
+            self.data (ndarray): 2D NumPy array where the first column is x-values 
+                                (linearly spaced between -10 and 10) and the second column 
+                                is the sine of these x-values.
+            self.frame_num (int): index of the current frame, initialized to 0.
```

Comment:
I think this is good information to have, but normally this notation style is used for function arguments. Can leave for now, but John might have a more specific notation that improv prefers.